conda/core/prefix_data.py

```diff
@@ -652,6 +654,26 @@ def _write_environment_state_file(self, state: dict[str, dict[str, str]]) -> Non
         env_vars_file = self.prefix_path / PREFIX_STATE_FILE
         env_vars_file.write_text(json.dumps(state, ensure_ascii=False))
 
+    def _ensure_no_reserved_env_vars(self, env_vars_names: Iterable[str]) -> None:
+        """
+        Ensure that the set of env_var_names does not contain any reserved env vars.
+        Will raise an OperationNotAllowed if a reserved env var is found.
+        """
+        invalid_vars = []
+        for var in RESERVED_ENV_VARS:
+            if var in env_vars_names:
+                invalid_vars.append(var)
+        if invalid_vars:
+            env_vars_file = self.prefix_path / PREFIX_STATE_FILE
+            warnings.warn(
+                f"Environment variable(s) '{' '.join(invalid_vars)}' are being modified. "
+                "These are reserved variables and the given configuration will not be applied "
+                "during environment activation. Setting these environment variables may "
+                "produce unexpected results.\n"
+                f"You may remove this invalid configuration by removing the "
+                f"'{' '.join(invalid_vars)}' key(s) from the file `{env_vars_file}`",
```

Comment:
```suggestion
                f"The given configuration will not be applied during environment activation as "
                 "the Environment variable(s) '{' '.join(invalid_vars)}' being modified are reserved variables."
                "Setting these environment variables may produce unexpected results.\n"
                f"You may remove this invalid configuration by removing the "
                f"'{' '.join(invalid_vars)}' key(s) from the file `{env_vars_file}`",
```

Reworded just a bit to make it less verbose.